src/wallet/wallet.cpp

```diff
@@ -6750,6 +6756,132 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+    auto recoverPath = GetOptionalArg("-recoverwallet");
+    if (recoverPath.has_value()) {
+        if (!fFirstRun) {
+            return UIError(_("Cannot recover over an existing wallet file."));
+        }
+        if (!fs::exists(recoverPath.value())) {
+            return UIError(_("Wallet recover file not found."));
+        }
+
+        // Parse file and search for "recovery_phrase="
+        std::ifstream recoverFile;
+        recoverFile.open(recoverPath.value(), std::ios::in);
+        assert(recoverFile.is_open());
+        SecureString line;
+        std::optional<SecureString> recoveryPhrase;
+        std::optional<Language> language;
+        std::optional<uint256> fingerprint;
+        uint32_t numAccounts = 1;
+
+        uint32_t lineNumber = 0;
+        uint32_t lastUsedLine;
+        while (recoverFile.good()) {
+            // Constrain parser to follow the multi-line format of z_exportwallet:
+            std::getline(recoverFile, line);
+            lineNumber++;
+
+            if (!line.empty()) {
+                auto recoverPos = line.find("recovery_phrase=");
+                if (recoverPos != std::string::npos) {
+                    // Check that file contains exactly one recovery phrase line
+                    if (recoveryPhrase.has_value()) {
+                        return UIError(_("The file should contain exactly one recovery phrase line."));
+                    }
+                    lastUsedLine = lineNumber;
+                    SecureString tmp = line.substr(recoverPos + 16, std::string::npos);
+                    boost::trim(tmp);
+                    if (tmp.front() == '\"' && tmp.back() == '\"') {
+                        tmp.erase(0, 1);
+                        tmp.pop_back();
+                    }
+                    recoveryPhrase = tmp;
+                }
+
+                auto languagePos = line.find("language=");
+                if (languagePos != std::string::npos) {
+                    // If a "language=" line appears, it must immediately follow
+                    // "recovery_phrase="
+                    if (!(recoveryPhrase.has_value() && lineNumber == (lastUsedLine + 1))) {
+                        return UIError(_("The 'language=' line must immediately follow the 'recovery_phrase=' line."));
+                    }
```

Comment:
I agree with @daira that this restriction could be relaxed; we just need to check the uniqueness of the language line, and then the `lastUsedLine` number isn't needed.